pints/tests/test_residuals_diagnostics.py

```diff
@@ -0,0 +1,199 @@
+#
+# Test the methods in pints.residuals_diagnostics
+#
+# This file is part of PINTS
+#  Copyright (c) 2017-2019, University of Oxford.
+#  For licensing informating, see the LICENSE file distributed with the PINTS
+#  software package.
+#
+import pints
+import pints.residuals_diagnostics
+import pints.toy as toy
+import unittest
+import numpy as np
+import matplotlib
+
+# Select matplotlib backend that doesn't require a screen
+matplotlib.use('Agg')
+
+
+class TestResidualsDiagnostics(unittest.TestCase):
+    """
+    Tests Pints residuals diagnostics methods.
+    """
+    def __init__(self, name):
+        super(TestResidualsDiagnostics, self).__init__(name)
+
+        # Create a single output optimization toy model
+        self.model1 = toy.LogisticModel()
+        self.real_parameters1 = [0.015, 500]
+        self.times1 = np.linspace(0, 1000, 100)
+        self.values1 = self.model1.simulate(self.real_parameters1, self.times1)
+
+        # Add noise
+        self.noise1 = 50
+        self.values1 += np.random.normal(0, self.noise1, self.values1.shape)
+
+        # Set up optimisation problem
+        self.problem1 = pints.SingleOutputProblem(
+            self.model1, self.times1, self.values1)
+        self.score1 = pints.SumOfSquaresError(self.problem1)
+        self.boundaries1 = pints.RectangularBoundaries([0, 200], [1, 1000])
+        self.x01 = np.array([0.5, 500])
+
+        # Run the optimisation
```

Comment:
Not sure about having this inside the test initialisation!

Reply:
You could just generate some random "found parameters" and speed things up _a lot_

Reply:
I used fixed found_parameters here, and I also replaced the MCMC chain with independent samples.